pkg/common/log/logger.go

```diff
@@ -7,151 +7,410 @@ SPDX-License-Identifier: Apache-2.0
 package log
 
 import (
+	"errors"
+	"fmt"
+	"os"
+	"strings"
 	"sync"
 
-	"github.com/hyperledger/aries-framework-go/pkg/internal/common/logging/metadata"
-	"github.com/hyperledger/aries-framework-go/spi/log"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
-//nolint:lll
 const (
-	// loggerNotInitializedMsg is used when a logger is not initialized before logging.
-	loggerNotInitializedMsg = "Default logger initialized (please call log.Initialize() if you wish to use a custom logger)"
-	loggerModule            = "aries-framework/common"
+	timestampKey  = "time"
+	levelKey      = "level"
+	moduleKey     = "logger"
+	callerKey     = "caller"
+	messageKey    = "msg"
+	stacktraceKey = "stacktrace"
 )
 
-// Log is an implementation of Logger interface.
-// It encapsulates default or custom logger to provide module and level based logging.
-type Log struct {
-	instance log.Logger
-	module   string
-	once     sync.Once
+// DefaultEncoding sets the default logger encoding.
+// It may be overridden at build time using the -ldflags option.
+//nolint:gochecknoglobals
+var DefaultEncoding = Console
+
+// Level defines a log level for logging messages.
+type Level int
+
+// String returns string representation of given log level.
+func (l Level) String() string {
+	switch l {
+	case DEBUG:
+		return "DEBUG"
+	case INFO:
+		return "INFO"
+	case WARNING:
+		return "WARN"
+	case ERROR:
+		return "ERROR"
+	case PANIC:
+		return "PANIC"
+	case FATAL:
+		return "FATAL"
+	default:
+		return fmt.Sprintf("Level(%d)", l)
+	}
 }
 
-// New creates and returns a Logger implementation based on given module name.
-// note: the underlying logger instance is lazy initialized on first use.
-// To use your own logger implementation provide logger provider in 'Initialize()' before logging any line.
-// If 'Initialize()' is not called before logging any line then default logging implementation will be used.
-func New(module string) *Log {
-	return &Log{module: module}
+// ParseLevel returns the level from the given string.
+func ParseLevel(level string) (Level, error) {
+	switch level {
+	case "DEBUG", "debug":
+		return DEBUG, nil
+	case "INFO", "info":
+		return INFO, nil
+	case "WARN", "warn", "WARNING", "warning":
+		return WARNING, nil
+	case "ERROR", "error":
+		return ERROR, nil
+	case "PANIC", "panic":
+		return PANIC, nil
+	case "FATAL", "fatal":
+		return FATAL, nil
+	default:
+		return ERROR, errors.New("logger: invalid log level")
```

Comment:
You can shorten the error text here to just `invalid log level`. Usually the error from this function will get wrapped by the caller, so it'll look like this:

```
level, err := log.ParseLevel("InvalidLevel")
if err != nil {
return fmt.Errorf("failed to parse log level: %w") // Wrap error
}
```

That wrapped error message will eventually print like this: `failed to parse log level: logger: invalid log level`. That extra `logger` text is pretty much redundant